account_invoice_inter_company/models/account_invoice.py

```diff
@@ -173,13 +173,15 @@ def _prepare_invoice_data(self, dest_inv_type, dest_journal_type,
         }
         dest_partner_data = self.env['account.invoice'].play_onchanges(
             dest_partner_data, ['partner_id'])
+        if not dest_partner_data.get('payment_term_id'):
+            dest_partner_data['payment_term_id'] = self.payment_term_id.id
         return {
             'name': self.name,
             'origin': _('%s - Invoice: %s') % (self.company_id.name,
                                                self.number),
             'type': dest_inv_type,
             'date_invoice': self.date_invoice,
-            'reference': self.reference,
+            'reference': _('Invoice: %s') % self.number,
```

Comment:
I think we should just put "self.number" in that field, without 'Invoice' prefix.